From 4eb9a15b6f4b5dab390bda8c8a979404d9263131 Mon Sep 17 00:00:00 2001
From: Christian Stewart <christian@paral.in>
Date: Sat, 23 Jul 2016 23:15:04 -0700
Subject: [PATCH 2/2] arm64: add -mpc-relative-literal-loads to avoid rela
 relocation errors

https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1533009

This patch fixes the error "Unsupported RELA relocation" when trying to
load modules on Arm64.

Signed-off-by: Christian Stewart <christian@paral.in>
---
 arch/arm64/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index df9d2fa..cf1ac29 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -38,6 +38,7 @@ CHECKFLAGS	+= -D__aarch64__
 
 ifeq ($(CONFIG_ARM64_ERRATUM_843419), y)
 KBUILD_CFLAGS_MODULE	+= -mcmodel=large
+KBUILD_CFLAGS_MODULE	+= $(call cc-option, -mpc-relative-literal-loads)
 endif
 
 # Default value
-- 
2.7.3

